RIOT-OS/RIOT
# boards/frdm: update openocd version requirements

### Contribution description

This PR drops the `Makefile.include` code and OpenOCD config related to older versions of OpenOCD (<0.10) for the NXP FRDM boards.

It also document the need for a built from source and more recent version of OpenOCD for KW41Z CPU models.




### Testing procedure

The following command should work with an OpenOCD version >= 0.10:
```
$ make BOARD=frdm-k64f -C examples/hello-world flash
```

The following command should work with a built from OpenOCD version that is earlier than https://github.com/ntfreak/openocd/commit/2c8602ed9f084d6680cec7d0ca1d5dc71c865a5f: 
```
$ make BOARD=frdm-kw41z -C examples/hello-world flash
$ make BOARD=usb-kw41z -C examples/hello-world flash
```




### Issues/PRs references

fixes #10857 (according to https://github.com/RIOT-OS/RIOT/issues/10857#issuecomment-458117903, only the doc is missing and this is added by 4c794c1)
